spec/lib/daemons/rails/monitoring_spec.rb

```diff
@@ -6,36 +6,28 @@
   [Daemons::Rails::Monitoring, Daemons::Rails::Monitoring.new(Rails.root.join('lib', 'daemons'))].each do |subject|
     it "should get list of controllers" do
       controllers = subject.controllers
-      controllers.should have(1).item
+      expect(controllers.length).to eq 1
       controller = controllers[0]
-      controller.path.should == Rails.root.join('lib', 'daemons', 'test_ctl')
-      controller.app_name.should == 'test.rb'
+      expect(controller.path).to eq Rails.root.join('lib', 'daemons', 'test_ctl')
+      expect(controller.app_name).to eq 'test.rb'
     end
 
     describe "using controllers" do
-      before :each do
-        @controller = Daemons::Rails::Controller.new(Rails.root.join('lib', 'daemons', 'test_ctl'))
-        Daemons::Rails::Monitoring.any_instance.stub(:controllers).and_return([@controller])
+      it "should start controller by name" do
+        subject.start('test.rb')
       end
 
       it "should return status for all controllers" do
-        @controller.should_receive(:run).with('status').and_return('test.rb: running [pid 10880]')
-        subject.statuses.should == {'test.rb' => :running}
-      end
-
-      it "should start controller by name" do
-        @controller.should_receive(:run).with('start')
-        subject.start('test.rb')
+        expect(subject.statuses).to eq({'test.rb' => :running})
       end
 
       it "should stop controller by name" do
-        @controller.should_receive(:run).with('stop')
         subject.stop('test.rb')
```

Comment:
It should check something after stop (i.e. job may stop update timestamp or counter in some file)

Reply:
I agree. Will try to add more tests for this scenario too